### ISSUE TITLE
Switch return error lint

### ISSUE THREAD

**Issue by Serginho** [palantir/tslint#1815]
**TypeScript Version:**  2.0.3 (tslint: 3.13.0)

**Code**

```ts
switch(foo){
case 0:
if(condition){
return 'something1';
else{
return 'something2';
}
case 1:
//Some code
}

```

**Expected behavior:**
Pass lint with no errors
**Actual behavior:**
TsLint error: expected a 'break' before 'case'. This's not happen if you use only return instead if clause, like:

```ts
// This pass lint inspection
case 0:
return 'something1';
case 1:
//Some code
```

**Comment by IllusionMH** [comment]
Seems like duplicate of #950

**Comment by ajafff** [comment]
This has been fixed some time ago by #2218

**Comment by omerson** [comment]
```
switch(foo){
case 0:
if(condition){
return 'something1';
}
return 'something2';
case 1:
//Some code
}
```
Should work fine
